java/test/jmri/jmrit/logix/LinkedWarrantTest.java

```diff
@@ -354,7 +356,16 @@ public void tearDown() {
         _sensorMgr.dispose();
         _sensorMgr = null;
         
-        JUnitUtil.clearShutDownManager(); // should be converted to check of scheduled ShutDownActions
+        if (InstanceManager.containsDefault(ShutDownManager.class)) {
+            ShutDownManager sm = InstanceManager.getDefault(jmri.ShutDownManager.class);
+            List<Runnable> rlist = sm.getRunnables();
+            while (!rlist.isEmpty()) {
+                if (rlist.get(0) instanceof jmri.jmrit.logix.WarrantShutdownTask) {
+                    sm.deregister((ShutDownTask)rlist.get(0));
+                    rlist = sm.getRunnables();
+                }
+            }
```

Comment:
If `rlist.get(0)` is not a `WarrantShutdownTask`, this will be an infinite loop. I suggest an `else` statement that throws an exception or breaks out of the loop.